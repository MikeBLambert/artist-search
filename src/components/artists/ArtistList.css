#artistList {
  background-color: black;

  #artistDisplay {
    background-color: lightgrey;
    border: 5px solid black;
    border-radius: 15px;
    margin-left: 25px;
    margin-right: 25px;
    box-shadow: 2px 2px lightgrey;
  }

}