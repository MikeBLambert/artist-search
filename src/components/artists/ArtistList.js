import React, { Component, Fragment } from 'react';
import Paging from '../paging/Paging';
import Artist from './Artist';
import Search from '../search/Search';
import { getArtists } from '../services/artistApi';

export default class ArtistList extends Component {
  state = {
    currentPage: 1,
    totalPages: 1,
    artists: [],
    searchQuery: ''
  };

  fetchArtist = () => {
    const { currentPage, searchQuery } = this.state;
    getArtists(currentPage, searchQuery)
      .then(({ artists, totalPages }) => {
        this.setState({ artists, totalPages });
      });
  }

  componentDidMount() {
    // this.fetchArtist();
  }

  handlePageUpdate = page => {
    this.setState({ currentPage: page }, () => {
      this.fetchArtist();
    });
  };

  handleSearchQueryUpdate = ({ target }) => {
    this.setState({ searchQuery: target.value }, () => {
      this.fetchArtist();
    });

  }

  render() {
    const { currentPage, totalPages, artists } = this.state;
    // const artistComponent = artists ? artists.map(artist => {
    //   return <Artist key={artist.id} id={artist.id} name={artist.name} />;
    // }) : <h1>No results found</h1>;

    const artistComponent = artists.map(artist => {
      return <Artist key={artist.id} id={artist.id} name={artist.name} />;
    });
    return (
      <Fragment>
        <Search updateSearchQuery={this.handleSearchQueryUpdate} />
        <Paging currentPage={currentPage} totalPages={totalPages} updatePage={this.handlePageUpdate} />
        {artistComponent}
      </Fragment>
    );
  }
}
