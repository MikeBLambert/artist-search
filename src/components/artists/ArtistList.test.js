import React from 'react';
import ArtistList from './ArtistList';
import { shallow } from 'enzyme';

describe('ArtistList', () => {
  it('renders an ArtistList component', () => {
    const wrapper = shallow(<ArtistList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has a paging component', () => {
    const wrapper = shallow(<ArtistList />);
    expect(wrapper.find('Paging')).toHaveLength(1);
  });

  it('has a search component', () => {
    const wrapper = shallow(<ArtistList />);
    expect(wrapper.find('Search')).toHaveLength(1);
  });
});
