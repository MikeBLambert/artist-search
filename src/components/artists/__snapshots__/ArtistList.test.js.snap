// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList renders an ArtistList component 1`] = `
<Fragment>
  <Search
    updateSearchQuery={[Function]}
  />
  <Paging
    currentPage={1}
    totalPages={1}
    updatePage={[Function]}
  />
</Fragment>
`;
